test/test_acknowledge.py

```diff
@@ -0,0 +1,887 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2015: Alignak team, see AUTHORS.txt file for contributors
+#
+# This file is part of Alignak.
+#
+# Alignak is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Alignak is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Alignak.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+This file test acknowledge.
+The acknowledge notifications are tested in test_notifications
+"""
+
+import time
+from alignak_test import AlignakTest
+
+
+class TestAcknowledges(AlignakTest):
+    """
+    This class test acknowledge
+    """
+
+    def test_ack_host_sticky_ds_dh(self):
+        """
+        Test host acknowledge with sticky when Down soft -> Down hard -> up
+
+        :return: None
+        """
+        self.print_header()
+        self.setup_with_file('cfg/cfg_default.cfg')
+
+        host = self.schedulers['scheduler-master'].sched.hosts.find_by_name("test_host_0")
+        host.checks_in_progress = []
+        host.act_depend_of = []  # ignore the router
+        host.event_handler_enabled = False
+
+        svc = self.schedulers['scheduler-master'].sched.services.find_srv_by_name_and_hostname("test_host_0",
+                                                                              "test_ok_0")
+        svc.checks_in_progress = []
+        svc.act_depend_of = []  # no hostchecks on critical checkresults
+
+        self.scheduler_loop(1, [[host, 0, 'UP'], [svc, 0, 'OK']])
+        time.sleep(0.1)
+        self.assertFalse(host.problem_has_been_acknowledged)
+        self.assert_actions_count(0)
+
+        self.scheduler_loop(1, [[host, 2, 'DOWN']])
+        time.sleep(0.1)
+        self.assertEqual("DOWN", host.state)
+        self.assertEqual("SOFT", host.state_type)
+
+        now = time.time()
+        cmd = "[{0}] ACKNOWLEDGE_HOST_PROBLEM;{1};{2};{3};{4};{5};{6}\n".\
+            format(now, host.host_name, 2, 0, 1, 'darth vader', 'normal process')
+        self.schedulers['scheduler-master'].sched.run_external_command(cmd)
+
```

Comment:
Why not testing immediately that: self.assertTrue(host.problem_has_been_acknowledged) ?